tests/integration/test_scale.py

```diff
@@ -0,0 +1,173 @@
+            for machine in data["machines"].values()
+            if machine["machine-status"]["current"] == "running"
+        )
+        - 1
+    )
+    while machines_available < machines_number:
+        await ops_test.juju("add-machine")
+        status = await ops_test.juju("status", "--format", "json")
+        data = json.loads(status[1])
+        machines_available = (
+            sum(
+                1
+                for machine in data["machines"].values()
+                if machine["machine-status"]["current"] == "running"
+            )
+            - 1
+        )
+        print("Available machines:", machines_available)
+        time.sleep(10)
+
+    for i in range(int(2000 / batch_number)):
+
+        status = await ops_test.juju("status", "--format", "json")
+        status_data = json.loads(status[1])
+        # we collect the machines but leave out machine "0"
+        # that should be in use by the bind-operator
+        machines = [x for x in status_data["machines"].keys() if x != "0"]
+        print("Available machines:", machines)
+
+        await asyncio.gather(
+            *[
+                deploy_any_charm(
+                    app,
+                    ops_test,
+                    model,
+                    i * batch_number + x,
+                    machines,
+                    [
+                        models.DnsEntry(
+                            domain="dns.test",
+                            host_label=f"admin{i * batch_number + x}",
+                            ttl=5,
+                            record_class="IN",
+                            record_type="A",
+                            record_data=(
+                                "1.1."
+                                f"{int(i * batch_number + x / 255) + 1}."
+                                f"{i * batch_number + x % 255 + 1}"
+                            ),
+                        ),
+                    ],
+                )
+                for x in range(batch_number)
+            ]
+        )
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.skip(reason="Scaling test")
+async def test_lots_of_record_requests(
+    app: ops.model.Application,
+    ops_test: OpsTest,
+    model: Model,
+):
+    """
+    arrange: build and deploy the charm.
+    act: deploy a lot of applications and relate them to the bind operator
+    assert:
+    """
+    application_number = 10
+    entries_per_application = 10000
+
+    for any_app_number in range(application_number):
+
+        entries = []
+
+        for entry_number in range(entries_per_application):
+            entries.append(
+                models.DnsEntry(
+                    domain="dns.test",
+                    host_label=f"admin{any_app_number}-{entry_number}",
+                    ttl=5,
+                    record_class="IN",
+                    record_type="A",
+                    record_data=f"1.1.{int(entry_number / 255) + 1}.{entry_number % 255 + 1}",
```

Comment:
maybe `%` instead of `/`?,

Reply:
Same as https://github.com/canonical/bind-operator/pull/43#discussion_r1732702413